ext/standard/var.c

```diff
@@ -782,13 +782,20 @@ static int php_var_serialize_try_add_sleep_prop(
 	if (Z_TYPE_P(val) == IS_INDIRECT) {
 		val = Z_INDIRECT_P(val);
 		if (Z_TYPE_P(val) == IS_UNDEF) {
-			zend_property_info *info = zend_get_typed_property_info_for_slot(Z_OBJ_P(struc), val);
-			if (info) {
-				zend_throw_error(NULL,
-					"Typed property %s::$%s must not be accessed before initialization (in __sleep)",
-					ZSTR_VAL(Z_OBJCE_P(struc)->name), ZSTR_VAL(error_name));
+			zend_property_info *prop_info = zend_get_typed_property_info_for_slot(Z_OBJ_P(struc), val);
+			if (!prop_info) {
+				return FAILURE;
 			}
-			return FAILURE;
+			zval *prop;
+			if ((prop_info->flags & ZEND_ACC_STATIC) != 0) {
```

Comment:
How would an instance property with a slot in the `zend_object` instance be static?
It looks like it's impossible, but if it's possible to be static, document how?

```
static inline zend_property_info *zend_get_property_info_for_slot(zend_object *obj, zval *slot)
{
	zend_property_info **table = obj->ce->properties_info_table;
	intptr_t prop_num = slot - obj->properties_table;
	ZEND_ASSERT(prop_num >= 0 && prop_num < obj->ce->default_properties_count);
	return table[prop_num];
}
// ...
/* Helper for cases where we're only interested in property info of typed properties. */
static inline zend_property_info *zend_get_typed_property_info_for_slot(zend_object *obj, zval *slot)
{
	zend_property_info *prop_info = zend_get_property_info_for_slot(obj, slot);
	if (prop_info && ZEND_TYPE_IS_SET(prop_info->type)) {
		return prop_info;
	}
	return NULL;
}
```

Reply:
`return ['a'];` from `__sleep()` where `$a` is a static property. This works already, since always.

Reply:
OK, got it, `zend_get_property_info_for_slot()` filters this case already, I'm now updating.